### ISSUE TITLE
Build CI/CD pipeline

### ISSUE THREAD

**Issue by minkj1992** [Nexters/myply#4]
# Action items
- github action
- gcp connect
- test before push

# GCP
> pricing

- [정리글](https://minkj1992.github.io/gcp_pricing/)

## tl;dr
**Just use Taiwan region**

## GCP network
> https://cloud.google.com/vpc/network-pricing

- Ingress ( users -> instance ) is free

![Screen Shot 2022-07-07 at 11 07 13 AM](https://user-images.githubusercontent.com/37536298/177675005-4ad16f45-dbc9-48a1-ab55-1cf7efeccfe1.png)

- egress ( kr -> taiwan $0.05 )

![Screen Shot 2022-07-07 at 11 08 19 AM](https://user-images.githubusercontent.com/37536298/177674992-11590792-af5d-454c-b0f5-4cde9032c33a.png)

## Cloud run
> https://cloud.google.com/run/pricing

1. Set cpu less than 1, memory 128MiB
2. Check CPU is only allocated during request processing
3. Execution env to set `First generation`
4. Set autoscaling  min number of instances field to 0. (max: 4~5?)

![Screen Shot 2022-07-07 at 10 17 45 AM](https://user-images.githubusercontent.com/37536298/177668849-966c3246-f473-49cd-a11f-46c078c21fcb.png)

- cloud run pricing by region, `FYI, seoul is 2nd grade.`
![Screen Shot 2022-07-07 at 10 42 21 AM](https://user-images.githubusercontent.com/37536298/177672190-eaa1e8e5-99f0-447f-8d05-30e6dc914486.png)

## Cloud SQL
> https://cloud.google.com/sql/pricing

- Set number of vCPU 1 (4로 설정하면 대략 하루 18,000원)
- Seoul이 vCPU가격이 다른 region에 비해서 훨씬 높다.  -> Select Taiwan

- Taiwan의 경우 us-central들과 가격 정책이 같다.
![Screen Shot 2022-07-07 at 10 28 07 AM](https://user-images.githubusercontent.com/37536298/177670809-bced7463-906c-4f8b-b70e-cee1c9d197bb.png)

- lowa
![Screen Shot 2022-07-07 at 10 29 18 AM](https://user-images.githubusercontent.com/37536298/177670842-0eeda4fb-d18a-4b07-b660-2523ed505276.png)

- seoul
![Screen Shot 2022-07-07 at 10 29 47 AM](https://user-images.githubusercontent.com/37536298/177670883-006897d1-1b58-4691-9d82-23a270b28e6f.png)

- **Data Egress(outboud) pricing**
  - Set `cloud run` region same as `cloud sql` 

![Screen Shot 2022-07-07 at 10 32 37 AM](https://user-images.githubusercontent.com/37536298/177671199-c2760c24-b0ab-4871-a9fc-69b6e34342ba.png)
